### PR TITLE
fix(ui): #1285 filter tool

### PR DESCRIPTION
**Pull request by gpazuch**
* adds filter service
* allow for multiple filters to be applied
* substring search for string properties: name, tags
* possible types of filter inputs: input, autocomplete, select, multiple select, checkbox

### CODE REVIEW

**Comment by gpazuch** [comment]
filter-tool
--current wip

File: ui/src/app/common/services/orb.service.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    private dataset: DatasetPoliciesService,
+    private group: AgentGroupsService,
+    private policy: AgentPoliciesService,
+    private sink: SinksService,
+  ) {
+    this.lastPollUpdate$ = new Subject<number>();
+    this.forceRefresh = new Subject<number>();
+    this.killPolling = new Subject<void>();
+
+    this.pollController$ = new BehaviorSubject<boolean>(PollControls.PAUSE);
```
Comment (gpazuch) [inline ui/src/app/common/services/orb.service.ts:105]: setting initial value to RESUME or calling next(RESUME) on this subject enables polling

**Comment by gpazuch** [comment]
image
image

**Review by gbottega — DISMISSED** [review]

File: ui/src/app/common/services/agents/agents.service.ts [hunk 2]
```diff
@@ -1,282 +1,236 @@
...
-      .catch(
-        err => {
-          this.notificationsService.error('Failed to Delete Agent',
-                                          `Error: ${err.status} - ${err.statusText}`);
-          return Observable.throwError(err);
-        },
-      );
+    return this.http
+      .delete(`${environment.agentsUrl}/${agentId}`)
+      .catch((err) => {
```
[the remark this answers is not in the document]
Comment (gpazuch) [reply ui/src/app/common/services/agents/agents.service.ts:121]: yup :)

**Review by manrodrigues — APPROVED** [review]
LGTM

**Review by manrodrigues — APPROVED** [review]
Ok

**Pull request merged**